core/src/main/java/com/exadel/aem/toolkit/bundle/customlists/datasource/ChildResourcesDatasource.java

```diff
@@ -0,0 +1,150 @@
+import com.adobe.granite.ui.components.PagingIterator;
+import com.adobe.granite.ui.components.ds.AbstractDataSource;
+import com.adobe.granite.ui.components.ds.DataSource;
+import com.adobe.granite.ui.components.ds.EmptyDataSource;
+import org.apache.commons.collections.iterators.TransformIterator;
+import org.apache.jackrabbit.commons.iterator.FilterIterator;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceWrapper;
+import org.apache.sling.api.scripting.SlingBindings;
+import org.apache.sling.api.scripting.SlingScriptHelper;
+import org.apache.sling.api.servlets.HttpConstants;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import javax.annotation.Nonnull;
+import javax.servlet.Servlet;
+import javax.servlet.ServletRequest;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static com.day.cq.commons.jcr.JcrConstants.JCR_CONTENT;
+import static com.day.cq.commons.jcr.JcrConstants.JCR_PRIMARYTYPE;
+import static com.day.cq.commons.jcr.JcrConstants.NT_FOLDER;
+import static com.day.cq.wcm.api.NameConstants.NN_TEMPLATE;
+import static com.day.cq.wcm.api.NameConstants.NT_PAGE;
+
+/**
+ * Retrieves all child pages under the current root path, which are either custom lists themselves,
+ * or folders that may contain lists inside. The result is then limited by 'offset' and 'limit' parameter values.
+ */
+@Component(
+    service = Servlet.class,
+    property = {
+        "sling.servlet.resourceTypes=aem-custom-lists/datasources/list",
+        "sling.servlet.methods=" + HttpConstants.METHOD_GET
+    }
+)
+public class ChildResourcesDatasource extends SlingSafeMethodsServlet {
+    private static final String TEMPLATE = "/conf/authoring-toolkit/settings/wcm/templates/custom-list";
+    private static final String PATH = "path";
+    private static final String OFFSET = "offset";
+    private static final String LIMIT = "limit";
+    private static final String REP_PREFIX = "rep:";
+    private static final String ITEM_RESOURCE_TYPE = "itemResourceType";
+    private static final String PATH_TO_JCR_CONTENT = String.format("/%s", JCR_CONTENT);
+
+    @Reference
+    private transient ExpressionResolver expressionResolver;
+
+    @Override
+    protected void doGet(@Nonnull SlingHttpServletRequest request, @Nonnull SlingHttpServletResponse response) {
+        ResourceResolver resolver = request.getResourceResolver();
+
+        SlingScriptHelper sling = getScriptHelper(request);
+        DataSource ds = EmptyDataSource.instance();
+
+        if (sling != null) {
+            ExpressionHelper ex = new ExpressionHelper(expressionResolver, request);
```

Comment:
Same here, especially. It's better to name it by purpose.

Reply:
Done.